megameklab/src/megameklab/printing/PrintSmallUnitSheet.java

```diff
@@ -183,4 +191,51 @@ private void printBottomTable(ReferenceTableBase table, PageFormat pageFormat) {
                 pageFormat.getImageableY() + pageFormat.getImageableHeight() * TABLE_RATIO + 3.0,
                 pageFormat.getImageableWidth() * TABLE_RATIO, pageFormat.getImageableHeight() * 0.2 - 3.0));
     }
+
+    private void addClusterChart() {
+        Element g = getSVGDocument().getElementById("unit_" + entities.size());
+        if (g == null) {
+            return;
+        }
+
+        var table = new ClusterHitsTable(this, entities, true);
+        if (!table.required()) {
+            return;
+        }
+
+        var uaa = new UserAgentAdapter();
+        var loader = new DocumentLoader(uaa);
+        var ctx = new BridgeContext(uaa, loader);
+        ctx.setDynamicState(BridgeContext.DYNAMIC);
+        new GVTBuilder().build(ctx, getSVGDocument());
+
+        var dims = SVGLocatableSupport.getBBox(getSVGDocument().getElementById("unit_0"));
+
+        var bbox = new Rectangle2D.Double(0, 10, dims.getWidth() + 5, dims.getHeight() - 20);
+
+        g.appendChild(table.createTable(bbox));
+    }
+
+    /**
+     * Determines if the supplied list of units fills the sheet or if there's room for more
+     * @param entities The list of entities to place on the sheet
+     * @param options The record sheet options, as reference tables can reduce available space
+     * @return {@code true} if no more entities can be printed on a single sheet
+     */
+    public static boolean fillsSheet(List<? extends Entity> entities, RecordSheetOptions options) {
+        var numTypes = entities.stream().map(Entity::getClass).distinct().count();
+        if (numTypes == 0) {
+            return false;
+        }
+        if (numTypes > 1) {
+            throw new IllegalArgumentException("Heterogeneous unit types are not supported");
+        }
+        if (entities.get(0) instanceof BattleArmor || entities.get(0) instanceof Protomech) {
```

Comment:
According to the style guide individual clauses in a conditional should be in separate parenthesis. i.e. `if `((cheese == 1) && (sausage == 2))` instead of `(cheese == 1 && sausage == 2)`